lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -191,26 +191,76 @@ class ConvertHasValueSemanticsOpsToValueTensors : public RewritePattern {
+      loc, Torch::NumberType::get(rewriter.getContext()), end, start);
+  Value div = rewriter.create<AtenDivOp>(loc, sub, step);
+  return rewriter.create<AtenCeilFloatOp>(loc, div);
+}
+
 class ReduceNonValueSemanticOps : public RewritePattern {
 public:
   ReduceNonValueSemanticOps(MLIRContext *context)
       : RewritePattern(MatchAnyOpTypeTag(), /*benefit=*/1, context) {}
   LogicalResult matchAndRewrite(Operation *op,
                                 PatternRewriter &rewriter) const override {
     Location loc = op->getLoc();
-    Operation *newOp;
+    MLIRContext *ctx = op->getContext();
     if (isa<AtenBernoulli_FloatOp>(op)) {
-      newOp = rewriter.create<ValsemVariantAtenBernoulliFloatOp>(
+      Operation *newOp = rewriter.create<ValsemVariantAtenBernoulliFloatOp>(
           loc, op->getResultTypes(), op->getOperands());
+      auto tensor =
+          rewriter.create<CopyToValueTensorOp>(loc, newOp->getResult(0));
+      createOverwriteTensorContents(rewriter, loc, tensor, op->getOperand(0));
+      rewriter.replaceOp(op, op->getOperand(0));
+      return success();
+    } else if (isa<AtenArangeStartOutOp>(op)) {
+      Value start = op->getOperand(0);
+      Value end = op->getOperand(1);
+      Value step = op->getOperand(2);
+      Value out = op->getOperand(3);
+
+      // `overwrite.tensor.contents` cannot change the tensor shape,
+      // so `out` tensor should have same num_elements with result tensor.
+      // It means that we don't support code like:
+      //   `x = torch.randn(12)`
+      //   `y = torch.arange(13, out=x)`
+      Value resultNumElements =
+          calculateArangeResultNumElements(rewriter, loc, start, end, step);
+      Value outNumElements = rewriter.create<AtenNumelOp>(loc, out);
+      Value eqOrNot =
+          rewriter.create<AtenEqIntOp>(loc, resultNumElements, outNumElements);
+      rewriter.create<RuntimeAssertOp>(
+          loc, eqOrNot,
+          rewriter.getStringAttr("`out` tensor should have the same "
+                                 "num_elements with result tenosr"));
+
+      auto dtype = rewriter.create<PrimDtypeOp>(loc, out);
+      auto device = rewriter.create<PrimDeviceOp>(loc, out);
+      auto shape = rewriter.create<AtenSizeOp>(
+          loc, Torch::ListType::get(Torch::IntType::get(ctx)), out);
+      auto none = rewriter.create<ConstantNoneOp>(loc);
+      auto cstFalse = rewriter.create<ConstantBoolOp>(loc, false);
+      Value newArange = rewriter.create<AtenArangeStartStepOp>(
+          loc, op->getResult(0).getType(), start, end, step, dtype,
+          /*dtype=*/none, device, /*pin_memory=*/none);
+      Value reshape = rewriter.create<AtenReshapeOp>(
+          loc, op->getResult(0).getType(), newArange, shape);
+      auto toDtype = rewriter.create<AtenToDtypeOp>(
+          loc, op->getResult(0).getType(), reshape, dtype,
+          /*non_blocking=*/cstFalse,
+          /*copy=*/cstFalse, /*memory_format=*/none);
```

Comment:
I would have expected specifying `dtype` in `AtenArangeStartStepOp` to take care of having the right output type. What error where you seeing without this? (it could be that `AtenArangeStartStepOp` is not handling the `dtype` argument properly, and if so, we should fix it there).

Reply:
Oh, you are right. I have removed the postpositional `AtenToDtypeOp`.